addons/account/models/res_partner_bank.py

```diff
@@ -63,6 +64,22 @@ def _check_allow_out_payment(self):
                 if not self.env.user.has_group('account.group_validate_bank_account'):
                     raise ValidationError(_('You do not have the right to trust or un-trust a bank account.'))
 
+    @api.depends('acc_number')
+    def _compute_alert_duplicate_bank_account(self):
+        """Add a warning if the bank account is already registered with another partner."""
+        for bank in self:
+            duplicate = self.env['res.partner.bank'].search([('acc_number', '=', bank.acc_number)], limit=1)
```

Comment:
`search()` in a loop is a big no-no :grimacing:

We need to get all the duplicates before (no `limit=1`) and we `mapped()` them by `acc_number`. When we are inside the loop we just need to check if we have a match in the duplicates.

(Or something like that.)